src/Compilers/CSharp/Portable/Binder/Binder_Statements.cs

```diff
@@ -3081,7 +3080,10 @@ internal BoundBlock CreateBlockFromExpression(CSharpSyntaxNode node, ImmutableAr
             }
             else
             {
-                statement = new BoundReturnStatement(syntax, refKind, BindToNaturalType(expression, diagnostics)) { WasCompilerGenerated = true };
+                // When binding for purpose of inferring the return type of a lambda, we do not require returned expressions (such as `default` or switch expressions) to have a natural type
+                var inferringLambda = this.ContainingMemberOrLambda is MethodSymbol method && (object)method.ReturnType == LambdaSymbol.ReturnTypeIsBeingInferred;
```

Comment:
>MethodSymbol [](start = 71, length = 12)

LambdaSymbol?

Reply:
I'm not sure. Other places that check for the sentinel return type (`LambdaSymbol.ReturnTypeIsBeingInferred`) just use `MethodSymbol`. That is sufficient to get the return type.

---
In reply to: [314114199](https://github.com/dotnet/roslyn/pull/37596#discussion_r314114199) [](ancestors = 314114199)